src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -331,6 +333,15 @@ def validate(self, node, options=None):
                 f"be references to arrays but found '{result.symbol}', "
                 f"'{matrix1.symbol}' and '{matrix2.symbol}'.")
 
+        # The arrays must not be potentially aliased
+        if any(var.symbol.is_potentially_aliased for var in
+               [matrix1, matrix2, result]):
+            raise TransformationError(
+                f"Expected result and operands of MATMUL IntrinsicCall to "
+                f"be references to arrays that are not potentially aliased "
+                f"but found '{result.symbol}', '{matrix1.symbol}' and "
+                f"'{matrix2.symbol}'.")
```

Comment:
probably worth adding "...symbol}' in '{node.debug_string()}'." to give more information.